streaming/src/main/scala/io/gearpump/streaming/executor/Executor.scala

```diff
@@ -179,11 +175,31 @@ class Executor(executorContext: ExecutorContext, userConf : UserConfig, launcher
 
     case query: QueryExecutorConfig =>
       sender ! ExecutorConfig(systemConfig)
+
+    case registered: TaskRegistered =>
+      taskToSessionId += registered.taskId -> registered.sessionId
+    case rejected: TaskRejected =>
+      tasks.get(rejected.taskId).foreach(_ ! PoisonPill)
```

Comment:
Do we want to combine taskToSessionId  with tasks, so that it is easier to manage consistency